File: math/fastex/fastex.java


public long recursiveFastex(int b, long e) {

    if(e == 0) return 1;
    long r = fastex(b,e >>1)
    return (e%2 == 1 ? r*r*b : r*r)
}


public long fastex(long b, long e) {

    long r = 1;

    while(0 <e) {
        if((e &1) == 1) r = b*r;
        b = b*b;
        e >>= 1;
    }
    return r;
}

File: math/fastex/fastex.js


function recursiveFastex(b,e) {

	if(!e) return 1;
	var r = recursiveFastex(b,e >>1)
	return (e &1 ? b*r*r : r*r);
}


function iterativeFastex(b,e) {

	var r = 1;

	for(; e; e >>= 1) {

		if(e &1) r = b*r;
		b = b*b;
	}
	return r;
}


